web/src/components/base/InputSection.tsx

```diff
@@ -20,143 +20,24 @@ export const Title = styled.label`
   font-weight: bold;
 `
 
-const GeneralInputStyles = css<{ submitted: boolean }>`
-  padding: 0.75rem;
-  background-clip: padding-box;
-  border: 1px solid ${props => props.theme.colors.textSecondaryColor};
-  ${props =>
-    props.submitted &&
-    css`
-      :invalid {
-        :focus {
-          outline-color: ${props => props.theme.colors.invalidInput}33;
-        }
-
-        border-width: 2px;
-        border-color: ${props => props.theme.colors.invalidInput};
-      }
-    `}
-`
-
-export const CompactTitle = styled.label`
-  position: relative;
-  padding: 0 4px;
-  top: 0.9rem;
-  left: 10px;
-  font-size: ${props => props.theme.fonts.decorativeFontSizeSmall};
-  background-color: ${props => props.theme.colors.backgroundColor};
-  width: fit-content;
-`
-
-const StyledTextArea = styled.textarea`
-  ${GeneralInputStyles};
-  border-radius: 0.2rem 0.2rem 0;
-  resize: vertical;
-  min-height: 60px;
-`
-
-const TextInput = styled.input`
-  ${GeneralInputStyles};
-  border-radius: 0.2rem;
-  background-clip: padding-box;
-  resize: none;
-`
-
-type InputProps = {
+type InputSectionProps = {
   id: string
-  value: string
-  submitted?: boolean
-  onChange: (input: string) => void
-  multiline?: boolean
-  numberOfLines?: number
-  maxLength?: number
-  required?: boolean
-}
-
-type InputSectionProps = (
-  | {
-      title: string
-      compact?: false
-      subtitle?: string
-      description?: string
-    }
-  | { title: string; compact: true }
-) &
-  ({ optional: true; showOptional?: boolean } | { optional?: false; showOptional?: false }) &
-  Omit<InputProps, 'required'>
-
-const Input = ({
-  id,
-  onChange,
-  numberOfLines = DEFAULT_MULTI_LINE_NUMBER,
-  value,
-  submitted = false,
-  multiline,
-  maxLength,
-  required,
-}: InputProps): ReactElement => {
-  if (multiline) {
-    return (
-      <StyledTextArea
-        id={id}
-        onChange={event => onChange(event.target.value)}
-        value={value}
-        rows={numberOfLines}
-        maxLength={maxLength}
-        required={required}
-        submitted={submitted}
-      />
-    )
-  }
-  return (
-    <TextInput
-      id={id}
-      type='text'
-      onChange={event => onChange(event.target.value)}
-      required={required}
-      value={value}
-      maxLength={maxLength}
-      submitted={submitted}
-    />
-  )
+  title: string
+  description?: string
+  showOptional?: boolean
+  children: React.ReactComponentElement<typeof InputComponent>
```

Comment:
Why do you pass the `Input` instead of rendering it directly here?

Reply:
To move the input logic to the input component instead of passing all Input props through.